tests/integration/models/reddit/test_rules.py

```diff
@@ -0,0 +1,139 @@
+            assert rule.short_name == "PRAW Test"
+
+    def test_get_rule_int(self):
+        with self.recorder.use_cassette("TestRule.test_get_rule_int"):
+            assert isinstance(self.subreddit.rules[0], Rule)
+
+    def test_get_rule_negative_int(self):
+        with self.recorder.use_cassette("TestRule.test_get_rule_int"):
+            assert isinstance(self.subreddit.rules[-1], Rule)
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_reorder_rules(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_reorder_rules"):
+            rule_list = list(self.subreddit.rules)
+            reordered = rule_list[2:3] + rule_list[0:2] + rule_list[3:]
+            rule_info = {rule.short_name: rule for rule in rule_list}
+            self.subreddit.rules.reorder(reordered)
+            for rule in self.subreddit.rules:
+                assert rule_info[rule.short_name] == rule
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_reorder_rules_no_reorder(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestRule.test_reorder_rules_no_reorder"
+        ):
+            rule_list = list(self.subreddit.rules)
+            assert self.subreddit.rules.reorder(rule_list) == rule_list
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_update_rule(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette("TestRule.test_update_rule"):
+            rule = self.subreddit.rules[0]
+            rule2 = rule.update(
+                description="Updated rule",
+                kind="link",
+                violation_reason="PUpdate",
+            )
+            assert rule.description != rule2.description
+            assert rule2.description == "Updated rule"
+            assert rule.kind != rule2.kind
+            assert rule2.kind == "link"
+            assert rule.violation_reason != rule2.violation_reason
+            assert rule2.violation_reason == "PUpdate"
+
+    @mock.patch("time.sleep", return_value=None)
+    def test_update_rule_short_name(self, _):
+        self.reddit.read_only = False
+        with self.recorder.use_cassette(
+            "TestRule.test_update_rule_short_name"
+        ):
+            rule = self.subreddit.rules[1]
+            rule2 = rule.update(
+                short_name="PRAW Update",
+                description="Updated rule",
+                kind="comment",
+                violation_reason="PUpdate",
+            )
+            assert rule != rule2
+            assert rule2.short_name == "PRAW Update"
+            assert rule.description != rule2.description
+            assert rule2.description == "Updated rule"
+            assert rule.kind != rule2.kind
+            assert rule2.kind == "comment"
+            assert rule.violation_reason != rule2.violation_reason
+            assert rule2.violation_reason == "PUpdate"
```

Comment:
Can we re-fetch the rules from the subreddit and make sure that there is no rule with the old short name? (i.e. show that the rule has been renamed and not just duplicated with a new name)

Reply:
That would be a good test to have.
```suggestion
            assert rule2.violation_reason == "PUpdate"
            for new_rule in self.subreddit.rules:
                assert new_rule.short_name != rule.short_name
```